### ISSUE TITLE
Standardize on GoogleUtilities Swizzler

### ISSUE THREAD

**Issue by paulb777** [firebase/firebase-ios-sdk#2591]
Standardize swizzler usage with the [GoogleUtilities Swizzler](https://github.com/firebase/firebase-ios-sdk/tree/master/GoogleUtilities) in FirebaseAuth and FirebaseMessaging.

**Comment by paulb777** [comment]
Contact @tejasd when ready to start working on a PR for this.

**Comment by maksymmalyhin** [comment]
Classes to be attended:
- [FIRAuthAppDelegateProxy](https://github.com/firebase/firebase-ios-sdk/blob/master/Firebase/Auth/Source/FIRAuthAppDelegateProxy.m)
- [FIRMessagingRemoteNotificationsProxy](https://github.com/firebase/firebase-ios-sdk/blob/master/Firebase/Messaging/FIRMessagingRemoteNotificationsProxy.m)

**Comment by maksymmalyhin** [comment]
Remaining work:
- Messaging: update swizzling of UNUserNotificationCenter ?
- GULAppDelegateSwizzler - add macOS support (#2706)
- Use private header in tests instead of [this](https://github.com/firebase/firebase-ios-sdk/blob/cd507af5590bb3eb1be5341b8a0f616202fe542d/Example/Messaging/Tests/FIRMessagingRemoteNotificationsProxyTest.m#L125)

**Comment by ryanwilson** [comment]
Is this something that has to go in with Firebase 6 or is just a nice to have?

**Comment by paulb777** [comment]
It either needs to be with Firebase 6 or reverted. We shouldn't have the release branch not match a major release.

**Comment by maksymmalyhin** [comment]
All 3 points are nice to have and are basically, nice improvements. I think, we should release the changes once #2743 is ready and create a separate ticket for those nice-to-have things.

**Comment by maksymmalyhin** [comment]
I created a separate ticket for the improvements: #2766.
